src/catalog/catalog_accessor.cpp

```diff
@@ -217,4 +220,8 @@ common::ManagedPointer<storage::BlockStore> CatalogAccessor::GetBlockStore() con
   return catalog_->GetBlockStore();
 }
 
+void CatalogAccessor::RegisterTempTable(table_oid_t table_oid, const common::ManagedPointer<storage::SqlTable> table) {
+  temp_tables_[table_oid] = table;
```

Comment:
Will this temp_table grow arbitrarily large?

Reply:
No its scope is limited to the scope of the catalogaccessor